### ISSUE TITLE
Using a Closure in BelongsToSelect->createOptionForm throws an error

### ISSUE THREAD

**Issue by kjlloydie** [laravel-filament/filament#2604]
### Package

filament/forms

### Package Version

v2.12.20

### Laravel Version

v9.14.1

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.5

### Bug description

The `Select:createFormOptions` function accepts a Closure for the form schema, but evaluating the closure appears to fail when evaluating the Livewire component ([filament/forms/src/Components/Component.php
:39](https://github.com/laravel-filament/filament/blob/d8e22f33bb004c6b659c7a86d7836bea8c3e42d8/packages/forms/src/Components/Component.php#L39))

### Steps to reproduce

From the docs, this works:
```php
BelongsToSelect::make('authorId')
    ->relationship('author', 'name')
    ->createOptionForm([
        Forms\Components\TextInput::make('name')
            ->required(),
        Forms\Components\TextInput::make('email')
            ->required()
            ->email(),
    ]),
```
but this does not:
```php
BelongsToSelect::make('authorId')
    ->relationship('author', 'name')
    ->createOptionForm(function () {
          return [
              Forms\Components\TextInput::make('name')
                  ->required(),
              Forms\Components\TextInput::make('email')
                  ->required()
                  ->email(),
          ];
      }),
```

### Relevant log output

```shell
https://flareapp.io/share/bP9OBO35#F80
```

**Comment by danharrin** [comment]
What's the use case? Not sure if we should even allow closures here or not.

**Comment by kjlloydie** [comment]
For my specific use case, I wanted to conditionally show the createOption action (obviously the closure would have actual logic in it 😉).

The [canCreateOption](https://github.com/laravel-filament/filament/blob/ff11d33d12276f6bb8c037acda0171783d228d49/packages/forms/src/Components/Select.php#L198) function in Select just returns true if the form is not null (although it doesn't look like that function is actually used when creating the option in [getCreateOptionAction](https://github.com/laravel-filament/filament/blob/ff11d33d12276f6bb8c037acda0171783d228d49/packages/forms/src/Components/Select.php#L155), though the same null check is done), so I figured seeing as [createOptionForm](https://github.com/laravel-filament/filament/blob/ff11d33d12276f6bb8c037acda0171783d228d49/packages/forms/src/Components/Select.php#L114) accepts a Closure, I'd just have that return null when I didn't want there to be a createOption action.

If there was a way to conditionally enable the createOption action without setting the form to null, I could just use that for my specific use case.

**Comment by ryangjchandler** [comment]
Sounds like we should have another method to determine if it's enabled or not.

**Comment by danharrin** [comment]
Fixed by #2621, I've refactored the entire action a little bit to be a little bit more flexible. You can now return `null` from the closure to disable the action conditionally.
